src/components/ui/coaching-sessions/tiptap-editor.tsx

```diff
@@ -1,266 +1,94 @@
 "use client";
 
-import { useEditor, EditorContent, ReactNodeViewRenderer } from "@tiptap/react";
-
-import Bold from "@tiptap/extension-bold";
-import BulletList from "@tiptap/extension-bullet-list";
-import Document from "@tiptap/extension-document";
-import Heading from "@tiptap/extension-heading";
-import Highlight from "@tiptap/extension-highlight";
-import Italic from "@tiptap/extension-italic";
-import ListItem from "@tiptap/extension-list-item";
-import OrderedList from "@tiptap/extension-ordered-list";
-import Paragraph from "@tiptap/extension-paragraph";
-import Strike from "@tiptap/extension-strike";
-import Text from "@tiptap/extension-text";
-import Underline from "@tiptap/extension-underline";
-import CodeBlockLowlight from "@tiptap/extension-code-block-lowlight";
-import {
-  Bold as BoldIcon,
-  Heading1,
-  Heading2,
-  Heading3,
-  Highlighter,
-  Italic as ItalicIcon,
-  Underline as UnderlineIcon,
-  List,
-  ListOrdered,
-  Strikethrough,
-  Braces,
-} from "lucide-react";
-import { Button } from "@/components/ui/button";
-import { forwardRef, useImperativeHandle } from "react";
-
-import { EditorRef } from "./coaching-notes";
-
-import css from "highlight.js/lib/languages/css";
-import js from "highlight.js/lib/languages/javascript";
-import ts from "highlight.js/lib/languages/typescript";
-import html from "highlight.js/lib/languages/xml";
-// Load all languages with "all" or common languages with "common"
-import { all, createLowlight } from "lowlight";
-
-// eslint-disable-next-line
-import CodeBlock from "@/components/ui/coaching-sessions/code-block";
-
+import { EditorProvider } from "@tiptap/react";
+import { TiptapCollabProvider } from "@hocuspocus/provider";
+import * as Y from "yjs";
+import { useEffect, useState } from "react";
+import { useCollaborationToken } from "@/lib/api/collaboration-token";
+import { useAuthStore } from "@/lib/providers/auth-store-provider";
+import { useCoachingSessionStateStore } from "@/lib/providers/coaching-session-state-store-provider";
+import { Extensions } from "@/components/ui/coaching-sessions/tiptap-editor/extensions";
+import { Toolbar } from "@/components/ui/coaching-sessions/tiptap-editor/toolbar";
+import { siteConfig } from "@/site.config";
 import "@/styles/styles.scss";
 
-const lowlight = createLowlight(all);
-
-lowlight.register("html", html);
-lowlight.register("css", css);
-lowlight.register("js", js);
-lowlight.register("ts", ts);
-
-interface TipTapProps {
-  editorContent: string;
-  onChange: (content: string) => void;
-}
+const useCollaborationProvider = (doc: Y.Doc) => {
+  const { currentCoachingSessionId } = useCoachingSessionStateStore(
+    (state) => state
+  );
+  const { userSession } = useAuthStore((state) => ({
+    userSession: state.userSession,
+  }));
+  const { jwt, isLoading, isError } = useCollaborationToken(
+    currentCoachingSessionId
+  );
+  const [isSyncing, setIsSyncing] = useState(true);
```

Comment:
Per my previous comment above, it'd be nice to surface this editor state up to the status label that sits under the editor's text box on the CoachingSession page.